### ISSUE TITLE
Error adding openid idp. x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by lightuponcloud** [minio/minio#16616]
## Expected Behavior
It should be possible to add openid idp using CLI

## Current Behavior
When I'm trying to add an IDP it returns error.

Command:
`mc --debug admin idp openid update client client_id=account client_secret="something" config_url="https://iam.hart902-iccr-148.ews.gic.example.com/realms/.well-known/openid-configuration" scopes="openid,groups"
`
Response:
```
{"Code":"XMinioAdminConfigBadJSON","Message":"Get \"https://iam.hart902-iccr-148.ews.gic.example.com/realms/.well-known/openid-configuration\": x509: certificate signed by unknown authority","Resource":"/minio/admin/v3/idp-config/openid/_","RequestId":"1743AB1062B5E450","HostId":"10555893-e690-4043-bca1-2f75bf331c2a"}
mc: <DEBUG> Response Time:  77.077787ms

mc: <ERROR> Unable to add OpenID IDP config to server. Get "https://iam.hart902-iccr-148.ews.gic.example.com/realms/.well-known/openid-configuration": x509: certificate signed by unknown authority
 (0) admin-idp-openid-add.go:91 cmd.mainAdminIDPOpenIDAddOrUpdate(..)
 Release-Tag:2.0.0-50 | Commit:DEVELOPMENT. | Host:data-object-storage-mn-mgt-5495589488-6f8kt | OS:linux | Arch:amd64 | Lang:go1.19.4 | Mem:70 MB/91 MB | Heap:70 MB/75 MB.

```

## Possible Solution
I tried adding ca.crt to CAs directory, but it did not work.
It should be possible to add trusted server certificates for IDP to trusted storage.

## Steps to Reproduce (for bugs)
1. `mc --debug admin idp openid update client client_id=account client_secret="something" config_url="https://iam.hart902-iccr-148.ews.gic.example.com/realms/.well-known/openid-configuration" scopes="openid,groups"
`

## Context
I'm trying to integrate external authentication / authorization with KeyCloak.
KeyCloak IDP is working behind load balancer, which has self-signed server key.

## Regression
We use a stable version of MinIO, 2.0.0-50

## Your Environment
 mc --version
mc version 2.0.0-50 (commit-id=DEVELOPMENT.GOGET)
Runtime: go1.19.4 linux/amd64
Copyright (c) 2015-0000 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>

* Server setup and configuration:
It is installed in kubernetes cluster

* Operating System and version (`uname -a`):
Linux eric-data-object-storage-mn-mgt-5495589488-6f8kt 5.3.18-57-default #1 SMP Wed Apr 28 10:54:41 UTC 2021 (ba3c2e9) x86_64 x86_64 x86_64 GNU/Linux

**Comment by vadmeste** [comment]
> We use a stable version of MinIO, 2.0.0-50

MinIO doesn't use this versioning schema, anyway, can you upgrade to the latest and check if the issues persists ?

Closing the issue. Feel free to reopen if the issue persists.

**Comment by lightuponcloud** [comment]
MinIO version:
It's RELEASE.2022-11-17T23-20-09Z (unmodified)

**Comment by harshavardhana** [comment]
> MinIO version: It's RELEASE.2022-11-17T23-20-09Z (unmodified)

Please upgrade to latest release of MinIO and make sure you have added the relevant public certs for your remote endpoint at `.minio/certs/CAs`

**Comment by lightuponcloud** [comment]
I found the following example code in master branch.
```
func parseDiscoveryDoc(ustr string) (DiscoveryDoc, error) {
        d := DiscoveryDoc{}
        req, err := http.NewRequest(http.MethodGet, ustr, nil)
        if err != nil {
                return d, err
        }
        clnt := http.Client{
                Transport: http.DefaultTransport,
        }

```

You do not use a custom x509 cert pool nor InsecureSkipVerify flag there. Do you use the same code for server side as well ?
It looks like MinIO server do not care of certificates that are in directory we pass in --certs-dir argument.

**Comment by lightuponcloud** [comment]
I found in openid.go this code:
`        openIDClientTransport := http.DefaultTransport
`
And in jwt.go:
```
        clnt := http.Client{
                Transport: transport,
        }
```
Where `transport http.RoundTripper`
Somewhere else you set
```
   var updateTransport http.RoundTripper = &http.Transport{
TLSClientConfig: &tls.Config{                        RootCAs:            globalRootCAs,

```

There are no signs that you are using the path provided by user in --certs-dir for outgoing connections to IDP.

**Comment by vadmeste** [comment]
> I found the following example code in master branch.

This is just an example

> There are no signs that you are using the path provided by user in --certs-dir for outgoing connections to IDP.

Using globalRootCAs means it is loading from --certs-dir (certs/CAs/ directory)

**Comment by caspereijkens** [comment]
My team mate suggested to use certsPath "/etc/ssl/certs/", this seems to solve the issue for us. (So change this line https://github.com/minio/minio/blob/master/helm/minio/values.yaml#L106)

**Comment by lightuponcloud** [comment]
It turned out minio server should be restarted after you put certificate to --certs-dir path
It should be mentioned in documentation.

**Comment by shkpk** [comment]
I landed to this thread while search for the exactly same problem I have. Trying to add idp having self signed SSL certificate and getting error.
I added ca-bundle.crt in /root/.minio/certs/CAs
```
# ls -al /root/.minio/certs/CAs
total 24
drwxr-xr-x. 1 root root 4096 Mar 28 10:37 .
drwxr-xr-x. 1 root root 4096 Mar 28 10:37 ..
-rw-rw-rw-. 1 root root 8126 Mar 28 10:37 ca-chain.pem
```
but minio keeps giving `Get "https://xxxxxxxx/realms/master/.well-known/openid-configuration": x509: certificate signed by unknown authority` error.
I used `quay.io/minio/minio:RELEASE.2023-03-24T21-41-23Z` image, and recreated a new image using Dockerfile
```
FROM quay.io/minio/minio:RELEASE.2023-03-24T21-41-23Z
RUN mkdir -p /root/.minio/certs/CAs/
COPY ./ca-chain.pem /root/.minio/certs/CAs/ca-chain.pem
```

can anyone help in this matter?

**Comment by vadmeste** [comment]
@shkpk this has been fixed and will be out in the next release.

**Comment by shkpk** [comment]
> @shkpk this has been fixed and will be out in the next release.

is there beta release available which I can use for now?

**Comment by vadmeste** [comment]
> is there beta release available which I can use for now?

No, but if you have golang installed, you just need to do `go install github.com/minio/minio@master`

**Comment by shkpk** [comment]
> > is there beta release available which I can use for now?
> 
> No, but if you have golang installed, you just need to do `go install github.com/minio/minio@master`

Thanks, I built minio binary and replaced it in container image of minio:RELEASE.2023-03-24T21-41-23Z and then used new container image and it worked perfectly fine when using local docker setup using docker compose.
However on production, we have minio-operator where we create tenant from operator console, there when I create tenant and gives Minio image to new container image from our artifactory, it creates tenant but adding idp still gives certificate error. 
I access container from shell, and confirmed that binary inside it is the one which I compiled
```
./minio -v
minio version DEVELOPMENT.GOGET (commit-id=DEVELOPMENT.GOGET)
Runtime: go1.19.7 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-0000 MinIO, Inc.
```
and certificates also exists in `/root/.minio/certs/CAs` directory.

Upon further checking, it is found that statefulset of tenant has following argument

```
containers:
      - args:
        - server
        - --certs-dir
        - /tmp/certs
```
I tried editing statefulset but it keeps reseting to this value of /tmp/certs
Can I get some help to find how can I get this run in tenant created through minio-operator?
